tree/dataframe/src/RDFInterfaceUtils.cxx

```diff
@@ -701,19 +704,19 @@ std::string JitBuildAction(const ColumnNames_t &bl, std::shared_ptr<RDFDetail::R
    std::stringstream createAction_str;
    createAction_str << "ROOT::Internal::RDF::CallBuildAction<" << actionTypeName;
    const auto columnTypeNames =
-      GetValidatedArgTypes(bl, customCols, tree, ds, actionTypeNameBase, /*vector2rvec=*/true);
+      GetValidatedArgTypes(cols, customCols, tree, ds, actionTypeNameBase, /*vector2rvec=*/true);
    for (auto &colType : columnTypeNames)
       createAction_str << ", " << colType;
    // on Windows, to prefix the hexadecimal value of a pointer with '0x',
    // one need to write: std::hex << std::showbase << (size_t)pointer
    createAction_str << ">(reinterpret_cast<std::shared_ptr<ROOT::Detail::RDF::RNodeBase>*>("
-                    << PrettyPrintAddr(prevNode) << "), {";
-   for (auto i = 0u; i < bl.size(); ++i) {
+                    << PrettyPrintAddr(prevNode) << "), new const char*[" << cols.size() << "]{";
```

Comment:
Did you try the alternative of generating the array of 'c-strings' in the stack rather than allocating with new?

Reply:
Yes it's a bit more annoying to generate the code in that case, because I need to instantiate a variable (can't be a temporary). The allocations are not a performance bottleneck so I kept the simpler code-generation logic.